Assets/Scripts/MeleeScript.cs

```diff
@@ -15,13 +20,101 @@ enum State
 
     State state;
 
-    protected override void Move()
+    private Vector3 target;
+
+    [SerializeField] private List<GameObject> alliesInRange;
+    private GameObject closestAlly;
+
+
+    private void Start()
     {
+        state = State.Chasing;
+        target = GameManager.Instance.PlayerTransform.position;
+    }
+
+    private void Update()
+    {
+        switch (state)
+        {
+            case State.Chasing:
+
+                target = DetectClosestAlly();
+                Move();
+                
+                float distance = Vector3.Distance(agent.transform.position, target);
+
+                if (distance < minAttackDistance)
+                {
+                    state = State.Attacking;
+                }
+
+                break;
+
+            case State.Attacking:
+
+                Attack();
+
+                break;
+
+            case State.Die:
+
+                break;
+        }
+        
+    }
 
+    protected override void Move()
+    {
+        Vector3 offset = target + (transform.position - target).normalized * (GameManager.Instance.playerColliderRadius + 1f);
+        agent.SetDestination(offset);
     }
 
     protected override void Attack()
     {
+        Debug.Log("Attacking");
+
+        agent.SetDestination(transform.position);
```

Comment:
Use agent.Stop() instead of pasing it's position

Reply:
Done